multi_partner_learning.py

```diff
@@ -356,10 +386,13 @@ def compute_collaborative_round_seqavg(self):
 
         # Starting model for each partner is the aggregated model from the previous collaborative round
         if is_very_first_minibatch:  # Except for the very first mini-batch where it is a new model
-            logger.debug(f"(seqavg) Very first minibatch, init a new model for the round")
-            model_for_round = self.generate_new_model()
+            if self.use_saved_weights:
+                logger.info(f"(seqavg) Very first minibatch of epoch n°{epoch_index}, init model with previous coalition model for each partner")
+            else:
+                logger.info(f"(seqavg) Very first minibatch of epoch n°{epoch_index}, init new model for each partner")
+            model_for_round = self.init_with_model()
```

Comment:
If i understand correctly, this method `compute_collaborative_round_seqavg()` is called at every epoch, which means at every epoch we start with a model from `init_with_model()`. And it looks like `init_with_model()` always returns the same model because the weights `self.init_model_from` are never updated. Am I missing something here ?

Reply:
🤓

Reply:
It's on the `is_very_first_minibatch`, my bad !